--- mozilla/toolkit/mozapps/installer/packager.mk~	2006-12-23 03:31:40.882927000 +0200
+++ mozilla/toolkit/mozapps/installer/packager.mk	2006-12-23 03:38:32.302235167 +0200
@@ -301,7 +301,7 @@
 # NOTE: this must be a tar now that dist links into the tree so that we
 # do not strip the binaries actually in the tree.
 	@echo "Creating package directory..."
-	@mkdir $(DIST)/$(MOZ_PKG_APPNAME)
+	@mkdir $(DESTDIR)$(MOZ_PKG_APPDIR)
 ifdef MOZ_PKG_MANIFEST
 	$(RM) -rf $(DIST)/xpt
 	$(PERL) -I$(topsrcdir)/xpinstall/packager -e 'use Packager; \
@@ -316,7 +316,7 @@
 	@cd $(DIST) && rsync -auv --copy-unsafe-links $(_APPNAME) $(MOZ_PKG_APPNAME)
 endif
 else
-	@cd $(DIST)/bin && tar $(TAR_CREATE_FLAGS) - * | (cd ../$(MOZ_PKG_APPNAME); tar -xf -)
+	cp -rfLp $(DIST)/bin/* $(DESTDIR)$(MOZ_PKG_APPDIR)
 endif # DMG
 endif # MOZ_PKG_MANIFEST
 ifndef PKG_SKIP_STRIP
@@ -348,7 +348,7 @@
 endif
 	@echo "Removing unpackaged files..."
 ifdef NO_PKG_FILES
-	cd $(DIST)/$(STAGEPATH)$(MOZ_PKG_APPNAME)$(_BINPATH); rm -rf $(NO_PKG_FILES)
+	cd $(DESTDIR)$(MOZ_PKG_APPDIR)$(_BINPATH); rm -rf $(NO_PKG_FILES)
 endif
 ifdef MOZ_PKG_REMOVALS
 	$(SYSINSTALL) $(MOZ_PKG_REMOVALS_GEN) $(DIST)/$(STAGEPATH)$(MOZ_PKG_APPNAME)$(_BINPATH)
